src/bsp/src/posix/src/BSP.cpp

```diff
@@ -0,0 +1,34 @@
+#include "BSP.h"
+#include <FreeRTOS.h>
+#include <task.h>
+#include <thread>
+
+BSP::BSP() : m_loopRate(0){};
```

Comment:
What happens if we use this constructor and thus the rosNodeHandle is null? And is this constructor overload used?

Reply:
This constructor is mainly to allow the declaration of an object of BSP without any dependency built for it. It is not meant to obtain a usable one, hence the need for a container. 

In short, it allow
```cpp
BSP bsp;
```

Reply:
It's a bad idea to have a public constructor that creates an unusable item

Reply:
Removed along placement new in Container